src/main/java/javax/jmdns/impl/DNSRecord.java

```diff
@@ -775,14 +749,14 @@ public boolean isSingleValued() {
         @Override
         boolean handleQuery(JmDNSImpl dns, long expirationTime) {
             ServiceInfoImpl info = (ServiceInfoImpl) dns.getServices().get(this.getKey());
-            if (info != null && (info.isAnnouncing() || info.isAnnounced()) && (_port != info.getPort() || !_server.equalsIgnoreCase(dns.getLocalHost().getName()))) {
+            if (info != null && (info.isAnnouncing() || info.isAnnounced()) && (port != info.getPort() || !server.equalsIgnoreCase(dns.getLocalHost().getName()))) {
```

Comment:
call be paranoid, but in my book neither the previous version nor the new version is null-safe with regards to server.
same in line 828/802

Reply:
You might be right, but it was never my intention to avoid NPE here. This line is just about renaming fields.

Reply:
I've put this to resolved, since below I've applied the other suggestion. Thanks for the hints!